src/worker-type.js

```diff
@@ -897,22 +934,14 @@ WorkerType.prototype.determineSpotBids = function(managedRegions, pricing, chang
       }
     }
 
-    if (!bidOptions.length) {
+    if (!cheapestBid) {
       log.error({
         workerType: this.workerType,
         priceTrace: priceTrace,
       }, 'could not create any bid');
       throw new Error('Could not create any bid for ' + this.workerType);
     }
 
-    // pick a random zone to bid into
-    let cheapest = _.sample(bidOptions);
```

Comment:
So you go back to always pick the cheapest region/AZ,

won't that fail, because of capacity issues which is no longer reflected in the price.

Please see: https://bugzilla.mozilla.org/show_bug.cgi?id=1422184#c7

I don't see this handling the "capacity-oversubscribed" signal, which we talked about on irc at the time.